extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java

```diff
@@ -36,36 +38,33 @@
 import java.text.ParseException;
 import java.time.Clock;
 import java.util.Optional;
-import java.util.concurrent.TimeUnit;
-import java.util.function.Supplier;
 
 public class DecentralizedIdentityService implements IdentityService {
-    private final Supplier<SignedJWT> verifiableCredentialProvider;
     private final DidResolverRegistry resolverRegistry;
     private final CredentialsVerifier credentialsVerifier;
     private final Monitor monitor;
+    private final PrivateKeyWrapper privateKey;
+    private final String issuer;
     private final Clock clock;
 
-    public DecentralizedIdentityService(Supplier<SignedJWT> vcProvider, DidResolverRegistry resolverRegistry, CredentialsVerifier credentialsVerifier, Monitor monitor, Clock clock) {
-        verifiableCredentialProvider = vcProvider;
+    public DecentralizedIdentityService(DidResolverRegistry resolverRegistry, CredentialsVerifier credentialsVerifier, Monitor monitor, PrivateKeyWrapper privateKey, String issuer, Clock clock) {
         this.resolverRegistry = resolverRegistry;
         this.credentialsVerifier = credentialsVerifier;
         this.monitor = monitor;
+        this.privateKey = privateKey;
+        this.issuer = issuer;
         this.clock = clock;
     }
 
     @Override
-    public Result<TokenRepresentation> obtainClientCredentials(String scope) {
-
-        var jwt = verifiableCredentialProvider.get();
+    public Result<TokenRepresentation> obtainClientCredentials(String scope, String audience) {
+        var jwt = VerifiableCredentialFactory.create(privateKey, issuer, audience, clock);
         var token = jwt.serialize();
-        var expiration = clock.millis() + TimeUnit.MINUTES.toMillis(10);
-
-        return Result.success(TokenRepresentation.Builder.newInstance().token(token).expiresIn(expiration).build());
+        return Result.success(TokenRepresentation.Builder.newInstance().token(token).build());
     }
 
     @Override
-    public Result<ClaimToken> verifyJwtToken(TokenRepresentation tokenRepresentation) {
+    public Result<ClaimToken> verifyJwtToken(TokenRepresentation tokenRepresentation, String audience) {
```

Comment:
Do we need the audience parameter here if the audience is also a part of the token? 
Is there a case when the audience is not a part of the token?

Reply:
The audience parameter is the audience we want to check against, ie the value of the audience claim we demand to be in the token